src/aiida/brokers/rabbitmq/broker.py

```diff
@@ -24,14 +29,16 @@
 class RabbitmqBroker(Broker):
     """Implementation of the message broker interface using RabbitMQ through ``kiwipy``."""
 
-    def __init__(self, profile: Profile) -> None:
+    def __init__(self, profile: Profile, loop=None) -> None:
         """Construct a new instance.
 
         :param profile: The profile.
         """
         self._profile = profile
-        self._communicator: 'RmqThreadCommunicator' | None = None
+        self._communicator: 'RmqThreadCommunicator | None' = None
```

Comment:
Right now we keep two communicator
```
Broker -> Communicator
       -> Coordinator -> Communicator
```
It could happen easily in the code that two communicators get out of sync (so reference to a different one). If we expose the Communicator in the Coordinator we could simplify it to
```
Broker -> Coordinator -> Communicator
```
and avoid this issue of two communicators. I am not sure how easy it is to replace the communicator in the coordinator, since no one is managing the old communicator anymore. So we would need to close the existing coordinator and raise an error if it cannot be closed.